File: ruby/bob/bob.rb
class Bob
  def self.hey(remark)
    return "Fine. Be that way!" if remark.scan(/\S/).empty?

    if question?(remark) && caps?(remark)
      "Calm down, I know what I'm doing!"
    elsif question?(remark)
      "Sure."
    elsif caps?(remark)
      "Whoa, chill out!"
    else
      "Whatever."
    end
  end

  def self.caps?(str)
    str == str.upcase && str.scan(/[A-Z]/).any?
  end

  def self.question?(str)
    str.strip[-1] == "?"
  end
end

File: elixir/bob/lib/bob.ex
defmodule Bob do
  def isNotGibberish(input) do
    String.upcase(input) !== String.downcase(input)
  end

  def isShouting(input) do
    String.upcase(input) == input && isNotGibberish(input)
  end

  def hey(rawInput) do
    input = String.trim(rawInput)
    cond do
      String.trim(input) == "" ->
        "Fine. Be that way!"
      String.at(input, -1) == "?" && isShouting(input) ->
        "Calm down, I know what I'm doing!"
      isShouting(input) ->
        "Whoa, chill out!"
      String.ends_with?(input, "?") ->
        "Sure."
      true ->
        "Whatever."
    end
  end
end